src/Stratis.Bitcoin/P2P/PeerConnectorConnect.cs

```diff
@@ -70,8 +70,41 @@ public override void OnStartConnect()
 
                 PeerAddress peerAddress = this.peerAddressManager.FindPeer(ipEndpoint);
                 if (peerAddress != null && !this.IsPeerConnected(peerAddress.Endpoint))
-                    await ConnectAsync(peerAddress).ConfigureAwait(false);
+                {
+                    // Nodes disallow connection to peers in same range to prevent sybil attacks.     
+                    if (ipEndpoint.Address.IsLocal())
```

Comment:
@thefazzer can we also please move all of this logic to a separate method? And we need to clean up the nested `if`s please :)

Effectively, we want to go:

```
if(!PeerIsFilteredFromRange(peerAddress)
    continue;

await this.ConnectAsync(peerAddress).ConfigureAwait(false);
```

Reply:
Just noting we pair programmed this and new approach has a lot more clarity.